packages/gamepads_platform_interface/lib/api/gamepad_controller.dart

```diff
@@ -9,14 +15,29 @@ class GamepadController {
   /// A user-facing, platform-dependant name for the gamepad controller.
   final String name;
 
+  final state = GamepadState();
+
+  StreamSubscription<GamepadEvent>? _subscription;
+
   GamepadController({
     required this.id,
     required this.name,
-  });
+    required GamepadsPlatformInterface plugin,
```

Comment:
Hmm, not too fond of this, it makes the API surface more complex. I'm not sure how to solve it nicely though.

Reply:
Ah, but maybe it doesn't matter, the user never creates `GamepadController`s themselves right?

Reply:
The problem is we want the gamepad controller class to be more than just a container, but to have functionality too. That really means it has to have access to the underlying APIs. The win32_gamepads package solves this by putting FFI code directly in their gamepad class, but this was the closest I got. 

And yes, users shouldn't be constructing these objects on their own, but rather getting them from `listGamepads()`, which means this will be handled automatically and shouldn't be breaking in practice.

Reply:
> And yes, users shouldn't be constructing these objects on their own

Yeah, I think this is fine then. :)